migen/build/lattice/icestorm.py

```diff
@@ -102,11 +106,24 @@ def __init__(self):
             "icepack {build_name}.txt {build_name}.bin"
         ]
 
+        self.nextpnr_yosys_template = [
+            "{read_files}",
+            "attrmap -tocase keep -imap keep=\"true\" keep=1 -imap keep=\"false\" keep=0 -remove keep=0",
+            "synth_ice40 -top top -json {build_name}.json",
+        ]
+
+        self.nextpnr_build_template = [
+            "yosys -q -l {build_name}.rpt {build_name}.ys",
+            "nextpnr-ice40 {pnr_pkg_opts} --pcf {build_name}.pcf --json {build_name}.json --asc {build_name}.txt",
+            "icetime {icetime_pkg_opts} -c {icetime_constraint} -t -p {build_name}.pcf -r {build_name}.tim {build_name}.txt",
```

Comment:
nextpnr has a built-in timing analyzer that's more precise than icetime; why not drop icetime? You also should pass the timing target to nextpnr as I believe it affects placement.